src/meta/src/hummock/manager/mod.rs

```diff
@@ -2443,30 +2443,64 @@ where
                     compactor_request_streams.push(future);
                 };
 
+            let mut event_loop_iteration_now = Instant::now();
             loop {
                 let shutdown_rx_shared = shutdown_rx_shared.clone();
 
+                // report
+                hummock_manager
+                    .metrics
+                    .compaction_event_loop_iteration_latency
+                    .observe(event_loop_iteration_now.elapsed().as_millis() as _);
+                event_loop_iteration_now = Instant::now();
+
                 tokio::select! {
                     _ = shutdown_rx_shared => {
                         return;
                     },
 
                     compactor_stream = compactor_streams_change_rx.recv() => {
                         if let Some((context_id, stream)) = compactor_stream {
-                                push_stream(context_id, stream, &mut compactor_request_streams);
+                            tracing::info!("compactor {} enters the cluster", context_id);
+                            push_stream(context_id, stream, &mut compactor_request_streams);
                         }
                     },
 
                     result = pending_on_none(compactor_request_streams.next()) => {
                         let mut compactor_alive = true;
                         let (context_id, compactor_stream_req) = result;
-                        let (event, stream) = match compactor_stream_req {
+                        let (event, create_at, stream) = match compactor_stream_req {
                             (Some(Ok(req)), stream) => {
-                                (req.event.unwrap(), stream)
+                                (req.event.unwrap(), req.create_at, stream)
                             }
-                            _ => continue,
+
+                            (Some(Err(err)), _stream) => {
+                                tracing::warn!("compactor {} leaving the cluster with err {:?}", context_id, err);
+                                hummock_manager.compactor_manager
+                                    .remove_compactor(context_id);
+                                continue
+                            }
+
+                            _ => {
+                                tracing::warn!("compactor {} leaving the cluster", context_id);
+                                hummock_manager.compactor_manager
+                                    .remove_compactor(context_id);
+                                continue
+                            },
                         };
 
+                        {
+                            let consumed_time_ms = SystemTime::now()
```

Comment:
Can we print a log if the consumed_time_ms is larger than expected (e.g. >500ms)?